cardinal/types/entity.go

```diff
@@ -1,3 +1,14 @@
 package types
 
+import "encoding/json"
+
 type EntityID uint64
+
+type EntityStateResponse []EntityStateElement
+
+type EntityStateRequest struct{}
```

Comment:
These types should just exist in the handler package. You already have `EntityStateElement` to use across packages, no?

Reply:
Yeah, I can organize it that way. Will do this tomorrow.

Reply:
This can't because Provider relies on EntityStateResponse. I can just kill the EntityStateResponse type type alltogether if needed. But for now I moved it to types/entity.go

Reply:
The provider relies on EntityStateResponse because you made `GetDebugState` returns `types.EntityStateResponse` when it could have just returned `[]EntityStateElement`. You are just getting confused by your own footguns. [](edited-with-Axolo)

Reply:
Seems like your stretched. I'll make a decision here to kill the type and merge it in, lmk if you want to reverse it.

Reply:
I already pushed the code and cleaned it myself.

Again, you are not actually reading the comment I left or asking clarification questions about it. You just ended up confusing yourself when what you did ends up being exactly what I'm describing. [](edited-with-Axolo)